src/Wave.Core/Consumers/PrimaryConsumer.cs

```diff
@@ -92,7 +92,7 @@ internal IHandlerResult ProcessMessage(RawMessage message)
             }
 
             // Invoke handler and get handler result
-            var handlerResult = this.PerformHandler(messageType, messageEnvelope);
+            var handlerResult = this.PerformHandler(messageType, messageEnvelope, message);
```

Comment:
It feels weird to pass both messageEnvelope and message here, as they are essentially different representation of the same thing.